### PR TITLE
[provcau-110] Update track processing widget

### PR DESCRIPTION
**Pull request by natasha-gaia**
provcau-110 Update track processing widget to fetch a maximum of 100 rows to stop time out.
Depending on how long it's been since you've processed a large amount of records in the task queue you may need to update the number of hours setting. 
image

### CODE REVIEW

**Comment by bri-gaia** [comment]
LGTM, tested locally.

**Review by bri-gaia — APPROVED** [review]
LGTM, tested locally.

File: app/widgets/trackProcessing/trackProcessingWidget.php [hunk 1]
```diff
@@ -77,15 +77,23 @@ public function renderWidget($ps_widget_id, &$pa_settings) {
...
+			", time() - (60*60*$pa_settings['hours']));
+
 			$qr_completed = $this->opo_db->query("
 				SELECT tq.*, u.fname, u.lname 
 				FROM ca_task_queue tq 
 				LEFT JOIN ca_users u ON u.user_id = tq.user_id 
-				WHERE tq.completed_on > ? 
+				WHERE tq.completed_on > ?
 				ORDER BY tq.completed_on desc
+				LIMIT 100
```
Comment (kehh) [inline app/widgets/trackProcessing/trackProcessingWidget.php]: please use $this->opn_limit which is already defined and configurable (and also defaults to 100).

I find it interesting that this limit is included in the while condition and we still end up with a timeout.

**Review by kehh — CHANGES_REQUESTED** [review]
tiny tweak sorry

File: app/widgets/trackProcessing/trackProcessingWidget.php [hunk 2]
```diff
@@ -77,15 +77,23 @@ public function renderWidget($ps_widget_id, &$pa_settings) {
...
 			
 			$vo_tq = new TaskQueue();
+			$qr_tq_total_count = $this->opo_db->query("
+				SELECT count(*) as _count
+				FROM ca_task_queue ctq 
+				WHERE ctq.completed_on > ?
+			", time() - (60*60*$pa_settings['hours']));
+
 			$qr_completed = $this->opo_db->query("
 				SELECT tq.*, u.fname, u.lname
```
Comment (bri-gaia) [inline app/widgets/trackProcessing/trackProcessingWidget.php]: @kehh this is pulling in _all_ task queue fields (`tq.*`), which the notes field for SolrIndexer in my local is around 3MB each. 

Selecting the specific fields that are used would be a lot more efficient, however I think limiting the query itself also makes sense as PHP needs to load the result of the whole query to be able to loop though it anyway, so I don't know if we care to pursue additional optimisations.

**Pull request merged**